OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -99,7 +99,7 @@ void RescanForTargets(Actor self)
 			targetCountdown = Info.TargetRescanInterval;
 
 			// If close enough, we don't care about other actors.
-			var target = self.World.FindActorsInCircle(self.CenterPosition, Info.IgnoreNoiseAttackRange).FirstOrDefault(x => x.Info.HasTraitInfo<AttractsWormsInfo>());
+			var target = self.World.FindActorsInCircle(self.CenterPosition, Info.IgnoreNoiseAttackRange).FirstOrDefault(x => x.Info.HasTraitInfo<WormFodderInfo>());
```

Comment:
@pchote suggested to use target types instead. This would become `.FirstOrDefault(x => attackTrait.HasAnyValidWeapons(Target.FromActor(x)));` then.